src/irmin/tree.ml

```diff
@@ -255,8 +287,11 @@ module Make (P : Private.S) = struct
           | Some x, Some y -> equal_contents x y
           | _ -> equal_hash (hash x) (hash y))
 
-    (* FIXME: fix compare *)
-    let t = Type.map ~equal:(Type.stage equal) v of_v (fun t -> t.v)
+    let compare (x : t) (y : t) = compare_hash (hash x) (hash y)
+
+    let t =
+      Type.map ~equal:(Type.stage equal) ~compare:(Type.stage compare) v of_v
```

Comment:
can you upstream that fix separately?

Reply:
Done in https://github.com/mirage/irmin/pull/1519.